### PR TITLE
Expansion and tuning of explainer css

### PR DESCRIPTION
**Pull request by jankrcal**
This PR expands styling of explainers, motivated by 3 different texts. Notably, it:
 - adds support for:
   - tl;drs for long texts
   - expander boxes,
   - series of texts (with an overview box),
   - wide figures (scrollable on mobile),
   - captions above figures,
   - sections with multiple columns (such as captions)
 - adjusts styling of TOC in longreads,
 - removes lightbox and figure links for mobile (as it does not work much)

### CODE REVIEW

**Comment by jankrcal** [comment]
Please consider allowing to override the default title for a tl;dr box (that might be handy for the GD explainer).

File: _includes/figure.html [hunk 1]
```diff
@@ -9,20 +12,24 @@
...
+      <div class="caption caption-below">
+        {{- include.caption | markdownify }}
+      </div>
     {%- endif %}
     {%- if include.source-text %}
-      Zdroj: {%- if include.source-url %}<a href="{{ include.source-url }}">{% endif %}
-        {{ include.source-text }}
-        {%- if include.source-url %}</a>{% endif %}
+      <div class="source">
+        Zdroj: {%- if include.source-url %}<a href="{{ include.source-url }}">{% endif %}
```
Comment (mgrabovsky) [inline _includes/figure.html:29]: We should probably localise the "Zdroj" later.

Comment (jankrcal) [reply _includes/figure.html:29]: Let me do it right away (as we anyway introduce new locale strings in this PR)

File: _includes/tldr.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div class="tldr">
+  <div class="title">{{ site.data.lang.text.tldr }}</div>
```
Comment (mgrabovsky) [inline _includes/tldr.html:2]: I propose a simple change to allow for a custom title:
```suggestion
  <div class="title">{{ include.title | default: site.data.lang.text.tldr }}</div>
```

File: assets/_scss/explainer.scss [hunk 3]
```diff
@@ -138,3 +153,58 @@
...
+
+.longread .tldr {
+    background-color: $c-extralight-blue;
+    border-radius: 1rem;
+    margin-block: 2rem;
+    padding: 1.5rem;
+    &, li {
+        line-height: 1.35;
+    }
+    .title {
```
Comment (mgrabovsky) [inline assets/_scss/explainer.scss:167]: I'm not thrilled about the look of the title. It looks a bit pale at the moment. I propose something in the following direction (`font-weight: bold; letter-spacing: .05em; text-transform: uppercase;`). What do you think?
image

Comment (jankrcal) [reply assets/_scss/explainer.scss:167]: Does it mean also removing the `opacity` param? I don't feel strongly about it so I've committed this change.

Comment (mgrabovsky) [reply assets/_scss/explainer.scss:167]: Yes, the opacity made it a bit too faint in my opinion.

**Comment by mgrabovsky** [comment]
Looks OK overall, although I haven't done a thorough review – there are too many changes at once.

There's one minor cosmetic bug: the horizontal scrollbar in wide figures is shown even if there's no content to scroll in Firefox (108.0 on Fedora). I believe changing `overflow-x: scroll` to `auto` should be enough.
Screenshot 2022-12-20 at 13-09-16 Jaké cesty mohou dovést Česko k bezemisní elektřině

**Comment by jankrcal** [comment]
Thanks! I know it's a monster PR, I am sorry about it.

(It's hard to aim for small PRs when you need to explore; I could've split it after the fact, though.)

**Comment by jankrcal** [comment]
Thanks for the proposal, fixed (on Mac, all scrollbars are hidden by default, so this was hard to spot for me :)

> There's one minor cosmetic bug: the horizontal scrollbar in wide figures is shown even if there's no content to scroll in Firefox (108.0 on Fedora). I believe changing `overflow-x: scroll` to `auto` should be enough.

Do you request any further changes? Should I coordinate with Katka on merging this?

**Review by mgrabovsky — APPROVED** [review]
Looks fine to me now.

I'll coordinate with Káťa.

**Comment by mgrabovsky** [comment]
I'll update and merge faktaoklimatu/web-cz#1141 as well if that's fine with you.

**Comment by jankrcal** [comment]
Thx!

**Pull request merged**